### PR TITLE
New practical example: Private Networking and Bastions

### PR DESCRIPTION
**Pull request by tigerlinux**
- Use of private networking with kops.
- Use of a bastion layer.
- How to make truly fully-available the bastion layer
- Show workaronds when using gossip-dns with bastions.

### CODE REVIEW

File: docs/examples/kops-tests-private-net-bastion-host.md [hunk 1]
```diff
@@ -0,0 +1,379 @@
+# USING KOPS WITH PRIVATE NETWORKING AND A BASTION HOST IN A HIGLY-AVAILABLE SETUP
+
+## WHAT WE WANT TO ACOMPLISH HERE ?.
```
Comment (chrislovecnm) [inline docs/examples/kops-tests-private-net-bastion-host.md]: Same comment about having a pre-flight check that spans the examples. Thoughts?

File: docs/examples/kops-tests-private-net-bastion-host.md [hunk 2]
```diff
@@ -0,0 +1,379 @@
...
+ip-172-20-111-44.ec2.internal   master  True
+ip-172-20-44-102.ec2.internal   node    True
+ip-172-20-53-10.ec2.internal    master  True
+ip-172-20-64-151.ec2.internal   node    True
+ip-172-20-74-55.ec2.internal    master  True
+
+Your cluster privatekopscluster.k8s.local is ready
+```
+
+## MAKING THE BASTION LAYER "HIGLY AVAILABLE".
```
Comment (chrislovecnm) [inline docs/examples/kops-tests-private-net-bastion-host.md]: We have duplication between the docs. Should we break the up into chapters?

File: docs/examples/kops-tests-private-net-bastion-host.md [hunk 3]
```diff
@@ -0,0 +1,379 @@
...
+
+If for any reason "godzilla" decides to destroy the amazon AZ that contains our bastion, we'll basically be unable to enter to our instances. Let's add some H.A. to our bastion layer and force amazon to deploy additional bastion instances on other availability zones.
```
Comment (chrislovecnm) [inline docs/examples/kops-tests-private-net-bastion-host.md]: Same note about Godzilla, he may be copyrighted, and thus we need a monster.

Comment (tigerlinux) [reply docs/examples/kops-tests-private-net-bastion-host.md]: jajaja.. yup you're right he may be copyrighted. I'll take all recommendations and requested changes into account. Putting a common "preflight" section and using chapters is a good idea. I'm going to work on the modifications during the next days, then send a new PR with all changes.

**Comment by blakebarnett** [comment]
This would be great to get in to the next release!

**Comment by tigerlinux** [comment]
Hi there,

My last PR already includes a more recent version of the changes on this one. Then, I'm closing this now!

**Pull request closed without merging**